tutorials/how_to_test_in_Flutter.md

```diff
@@ -0,0 +1,217 @@
+# Test in Flutter
+
+The Process manual states that code should be properly tested before a user story is done. This guide describes how you test in Flutter, and what the test criteria is in both blocs and screens/ widgets. 
+
+## Test location
+All tests have to be located in the "test" folder. All Weekplanner files should have a corresponding file in the test folder where all tests related to the file are located. The test files should be in the folder that corresponds to the content of the file. For example, screen testes should be in the screen folder.
+
+## Test basics
+A test file always contain a main function. The main function is placed directly into the file, and not in any type of class, and all tests are written in the main function.
+
+### Setup
+When several tests are run, they often share the same setup. You can automate the setup by creating a setUp() function in the main function. The setup function is then called every time a new test is run. The main function could look like this:
+
+``` Dart
+void main() {
+    SomeClass a;
+    SomeOtherClass b;
+    BlocType bloc;
+
+    setUp((){
+        a = <Some value>;
+        b = <Some other value>;
+        bloc = bloc(some API);
+    });
+
+    //The tests go here
+}
+```
+
+The variables you use in the setUp() function should be declared in the beginning of the main function, right before the setUp() function. 
+
+### Expect
+When you run a test you need to compare values to see if the test is successful. In Flutter this is done with an expect function. This is written:
+``` Dart
+    expect(value, expected_value);
+```
+
+There can be several expects inside the same test. Aside from an actual value, you can use a lot of different find predicate. The most commonly used is the following:
+``` Dart
+expect(<variable>, isTrue);
+expect(<variable>, isFalse);
+expect(<variable>, equals(<value>));
+expect(<variable>, findsNothing);
+expect(<variable>, findsNWidgets(<number>));
+expect(<variable>, findsWidgets);
+expect(<variable>, findsOneWidget);
+expect(<variable>, find);
+expect(<variable>, false);
+expect(<variable>, true);
+```
+## Writing tests
+
+There are a significant difference between a widget/screen test and a standard class like a bloc. The tests for these two are explained separately. 
+
+### Standard classes
```

Comment:
What exactly is a standard class, could you define it more clearly?
Is this how you are suppose to test bloc's or is that section missing?